pkg/hubble/parser/threefour/parser.go

```diff
@@ -147,13 +147,19 @@ func (p *Parser) Decode(data []byte, decoded *pb.Flow) error {
 	if len(data) < packetOffset {
 		return fmt.Errorf("not enough bytes to decode %d", data)
 	}
+	data = data[packetOffset:]
 
 	p.packet.Lock()
 	defer p.packet.Unlock()
 
-	err := p.packet.decLayer.DecodeLayers(data[packetOffset:], &p.packet.Layers)
-	if err != nil {
-		return err
+	// Since v1.1.18, DecodeLayers returns a non-nil error for an empty packet, see
+	// https://github.com/google/gopacket/issues/846
+	// TODO: reconsider this check if the issue is fixed upstream
+	if len(data) > 0 {
+		err := p.packet.decLayer.DecodeLayers(data, &p.packet.Layers)
```

Comment:
Actually, on closer inspection, I think we need to truncate `p.packet.Layers` somehow if it is not passed through DecodeLayers to avoid accidental reuse.

Reply:
Thanks, good point. This is what the suggested fix in google/gopacket#847 also does.